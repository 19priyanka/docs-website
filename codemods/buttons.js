const visit = require('unist-util-visit');
const {
  removeAttribute,
  addAttribute,
  hasClassName,
  findAttribute,
} = require('./utils/mdxast');

const VARIANTS = ['btn-primary'];

const buttons = () => (tree, file) => {
  if (file.path === './src/content/docs/agents/go-agent/index.mdx') {
    console.dir(tree, { depth: 4 });
  }
  let addImport = false;
  visit(
    tree,
    (node) => {
      return hasClassName('btn', node);
    },
    (a) => {
      addImport = true;
      const className = findAttribute('className', a);
      const href = findAttribute('href', a);

      const variant = className
        .split(/\s+/)
        .find(
          (className) =>
            className.startsWith('btn-') && VARIANTS.includes(className)
        )
        .replace('btn-', '');

      if (href) {
        a.attributes.push({
          name: 'as',
          value: {
            type: 'mdxValueExpression',
            value: 'Link',
          },
          type: 'mdxAttribute',
        });
        addAttribute('to', href, a);
      }

      a.name = 'Button';
      addAttribute('variant', variant, a);
      removeAttribute('className', a);
      removeAttribute('href', a);
    }
  );

  // We are assuming all pages have frontmatter

  if (addImport) {
    tree.children.splice(1, 0, {
      type: 'import',
      value: `import { Link } from 'gatsby'`,
    });
  }
};

module.exports = buttons;
