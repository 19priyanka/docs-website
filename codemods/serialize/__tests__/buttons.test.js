import buttons from '../buttons';
import createProcessor from '../../../scripts/mdx-to-html';

const input = `
<ButtonLink
  to="/docs/agents/c-sdk/table-of-contents"
  variant="normal"
>
  View all C SDK docs
</ButtonLink>

`;

test('serialize button to html', async () => {
  const processor = createProcessor({ codemods: [buttons] });
  const file = await processor.process(input);
  expect(file.contents).toMatchSnapshot();
});
