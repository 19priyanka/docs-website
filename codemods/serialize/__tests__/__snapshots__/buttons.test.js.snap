// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialize button to html 1`] = `"<div data-prop=\\"W3sidHlwZSI6Im1keEF0dHJpYnV0ZSIsIm5hbWUiOiJ0byIsInZhbHVlIjoiL2RvY3MvYWdlbnRzL2Mtc2RrL3RhYmxlLW9mLWNvbnRlbnRzIn0seyJ0eXBlIjoibWR4QXR0cmlidXRlIiwibmFtZSI6InZhcmlhbnQiLCJ2YWx1ZSI6Im5vcm1hbCJ9XQ==\\" data-component=\\"ButtonLink\\"><p>View all C SDK docs</p></div>"`;
